File: trail2/잔해물을 덮기 위한 사각형의 최소 넓이/minimum-area-of-rectangle-to-cover-debris.java
import java.io.*;
import java.util.*;

public class Main {
    static void paint(int[][] grid, int[] pos, boolean isCover) {
        for (int i = pos[0]; i < pos[2]; i += 1) {
            for (int j = pos[1]; j < pos[3]; j += 1) {
                grid[i][j] = isCover ? 0 : 1;
            }
        }
    }
    public static void main(String[] args) throws IOException {
        // Please write your code here.
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringTokenizer st;

        final int MAX_W = 2000;
        final int OFFSET = 1000;

        int[][] grid = new int[MAX_W + 1][MAX_W + 1];

        for (int i = 0; i < 2; i += 1) {
            st = new StringTokenizer(br.readLine());
            int[] pos = new int[4];

            for (int j = 0; j < 4; j += 1) {
                pos[j] = Integer.parseInt(st.nextToken()) + OFFSET;
            }

            if (i == 1) paint(grid, pos, true);
            else paint(grid, pos, false);
        }

        int[] coverPos = new int[4];
        coverPos[0] = Integer.MAX_VALUE;
        coverPos[1] = Integer.MAX_VALUE;

        int cnt = 0;

        for (int i = 0; i < MAX_W; i += 1) {
            for (int j = 0; j < MAX_W; j += 1) {
                if (grid[i][j] == 1) {
                    coverPos[0] = Math.min(coverPos[0], i);
                    coverPos[1] = Math.min(coverPos[1], j);
                    coverPos[2] = Math.max(coverPos[2], i);
                    coverPos[3] = Math.max(coverPos[3], j);
                    cnt += 1;
                }
            }
        }
        if (cnt == 0) {
            bw.write(String.valueOf(0));
        }
        else {
            bw.write(String.valueOf((coverPos[3] - coverPos[1] + 1) * (coverPos[2] - coverPos[0] + 1)));
        }
        bw.flush();
        bw.close();
    }
}

File: trail2/잔해물을 덮기 위한 사각형의 최소 넓이/minimum-area-of-rectangle-to-cover-debris.py
x1, y1, x2, y2 = [0] * 2, [0] * 2, [0] * 2, [0] * 2
x1[0], y1[0], x2[0], y2[0] = map(int, input().split())
x1[1], y1[1], x2[1], y2[1] = map(int, input().split())

# Please write your code here.
OFFSET = 1000
grid = [[0 for _ in range(OFFSET * 2 + 1)] for _ in range(OFFSET * 2 + 1)]

for i in range(2):
    for row in range(x1[i], x2[i]):
        for col in range(y1[i], y2[i]):
            grid[row + OFFSET][col + OFFSET] = 1 if i == 0 else 0

min_row = 10000
min_col = 10000
max_row = 0
max_col = 0
for row in range(OFFSET * 2 + 1):
    for col in range(OFFSET * 2 + 1):
        if grid[row][col] == 1:
            min_row = min(min_row, row)
            max_row = max(max_row, row)
            min_col = min(min_col, col)
            max_col = max(max_col, col)

if max_col == 0 or max_row == 0:
    print(0)
else :
    print((max_col - min_col + 1) * (max_row - min_row + 1))